tests/functionaltests/UXTestHelpers.mm

```diff
@@ -106,4 +114,37 @@
     }
 }
 
+//
+// UXEvent methods
+//
+void UXEvent::Set() {
+    [_condition lock];
+    _signaled = true;
+    [_condition signal];
+    [_condition unlock];
+}
+
+void UXEvent::Reset() {
+    [_condition lock];
+    _signaled = false;
+    [_condition unlock];
+}
+
+bool UXEvent::WaitFor(int timeOutInSeconds) {
+    [_condition lock];
+
+    if (!_signaled) {
+        _signaled = [_condition waitUntilDate:[NSDate dateWithTimeIntervalSinceNow:timeOutInSeconds]] ? true : false;
+    }
+
+    bool timedOut = !_signaled;
+    [_condition unlock];
+
+    if (_eventType == AutoReset) {
+        Reset();
+    }
+
+    return timedOut;
```

Comment:
yes, otherwise this is just plain confusing. thanks!

---
In reply to: [98278040](https://github.com/Microsoft/WinObjC/pull/1855#discussion_r98278040) [](ancestors = 98278040)